ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py

```diff
@@ -70,10 +73,266 @@ def _check_query(query):
+    }
+
+
+@st.composite
+def _helper_tensorarray_unstack(draw):
+    shape = draw(helpers.get_shape(min_num_dims=1))
+    size = draw(st.integers(1, 10))
+    dynamic_size = draw(st.booleans()) if size >= shape[0] else True
+    dtype = draw(helpers.get_dtypes(full=False, prune_function=False))[0]
+    tensor = draw(helpers.array_values(dtype=dtype, shape=shape))
+    kwargs = {"dtype": dtype, "size": size, "dynamic_size": dynamic_size}
+    return tensor, kwargs
+
+
 # --- Main --- #
 # ------------ #
 
 
+@given(l_kwargs=_helper_random_tensorarray())
+def test_tensorarray_close(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs)
+    ta_frontend.close()
+    ta.close()
+    assert np.array(ta_frontend.size()) == 0
+    assert np.array(ta.size()) == 0
+
+
+@given(l_kwargs=_helper_random_tensorarray(fn="concat"))
+def test_tensorarray_concat(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs)
+    helpers.value_test(
+        ret_np_from_gt_flat=ta.concat().numpy().flatten(),
+        ret_np_flat=np.array(ta_frontend.concat()).flatten(),
+        backend=backend_fw,
+    )
+
+
+@given(l_kwargs=_helper_random_tensorarray())
+def test_tensorarray_dtype(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs)
+    assert ta.dtype == ta_frontend.dtype.ivy_dtype
+
+
+@given(l_kwargs=_helper_random_tensorarray(fn="gather"))
+def test_tensorarray_gather(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs[:2])
+    *_, indices = l_kwargs
+    helpers.value_test(
+        ret_np_from_gt_flat=ta.gather(indices).numpy().flatten(),
+        ret_np_flat=np.array(ta_frontend.gather(indices)).flatten(),
+        backend=backend_fw,
+    )
+
+
+@given(l_kwargs=_helper_random_tensorarray())
+def test_tensorarray_read(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs)
+    id_read, _ = l_kwargs
+    for id, read in id_read:
+        helpers.value_test(
+            ret_np_from_gt_flat=ta.read(id).numpy().flatten(),
+            ret_np_flat=np.array(ta_frontend.read(id)).flatten(),
+            backend=backend_fw,
+        )
+
+
+@given(l_kwargs=_helper_random_tensorarray(fn="scatter"))
+def test_tensorarray_scatter(
+    l_kwargs,
+    backend_fw,
+):
+    id_read, _ = l_kwargs
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs, "scatter")
+    for id, read in id_read:
+        helpers.value_test(
+            ret_np_from_gt_flat=ta.read(id).numpy().flatten(),
+            ret_np_flat=np.array(ta_frontend.read(id)).flatten(),
+            backend=backend_fw,
+        )
+
+
+@given(l_kwargs=_helper_random_tensorarray())
+def test_tensorarray_size(
+    l_kwargs,
+    backend_fw,
+):
+    ta, ta_frontend = _helper_init_tensorarray(backend_fw, l_kwargs)
+    id_read, _ = l_kwargs
```

Comment:
I think we don't need `id_read`, right? I don't see it used anywhere in the test.

Reply:
Yes that's not required, I had done some refactoring and somehow missed removing it. Thanks for catching it.